examples/custom-headers-and-txs-node/src/primitives.rs

```diff
@@ -0,0 +1,14 @@
+use crate::header::CustomHeader;
+use alloy_consensus::Block;
+use reth_primitives_traits::NodePrimitives;
+
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct CustomNodePrimitives;
+
+impl NodePrimitives for CustomNodePrimitives {
+    type BlockHeader = CustomHeader;
+    type Block = Block<reth_primitives::TransactionSigned, CustomHeader>;
```

Comment:
Integration wise this looks right @mattsse ?i guess we want to add some helpers for extending existing types also?

Reply:
I think for this example we can simply go with a block type that only consists of transactions